lightpath/ui/widgets.py

```diff
@@ -177,21 +180,52 @@ def add_signal(self, signal):
         self.command_layout.insertWidget(count - 1, label, 0, Qt.AlignHCenter)
         self.command_layout.insertWidget(count, widget, 0, Qt.AlignHCenter)
 
+    def get_state_color(self) -> QColor:
+        """
+        Determine the icon color given the device state and path status
+        If device is an impediment: StateColor.Blocking
+        If device is removed: StateColor.Removed
+        If device is inserted, not blocking (mirrors): StateColor.HalfRemoved
+        If device is unknown / errored: StateColor.Error
```

Comment:
Docstring here is left over from a previous iteration of the code where the state colors were an enum, I think?

Reply:
Ah right, they're dictionary access now.